macros/src/main/scala/gestalt/macros/Monadless.scala

```diff
@@ -14,13 +14,19 @@ trait Monadless[Monad[_]] {
   */
 
   def lift[T](body: T)(implicit m: toolbox.WeakTypeTag[Monad[_]]): Monad[T] = meta {
+    import toolbox._
+
     val tree = Transformer(toolbox)(this, body)
+
+    val unliftSym = m.tpe.methodIn("unlift").headOption.map(_.symbol)
+    def isUnlift(tp: Type) = tp.denot.map(_.symbol) == unliftSym
+
+
     toolbox.traverse(tree) {
-      case tree @ q"$pack.unlift[$t]($v)" =>
-        toolbox.error("Unsupported unlift position", tree)
-      case tree @ q"unlift[$t]($v)" =>
+      case tree @ Apply(fun, args) if fun.hasType && isUnlift(fun.tpe) =>
```

Comment:
Now I've change the original crappy implementation by the author to more solid implementation. As can be seen, now using extractors in patterns is much simpler and cleaner than quasiquote-based patterns.

This is intended to be another case to support the hypothesis that `extractor-based patterns are better than quasiquote-based patterns`.

Another observation is that in order to do it the correct way, the concept `Symbol` seems necessary for checking a particular method (maybe polymorphic) is used. Nevertheless, I'm convinced meta-programmers should work with denotations most of the time, as it really simplifies working with path-dependent types and refinement types.